edx_notifications/stores/sql/models.py

```diff
@@ -0,0 +1,256 @@
+"""
+Django ORM models to support the Notification Store SQL backend
+"""
+
+from django.db import models
+
+from model_utils.models import TimeStampedModel
+
+from edx_notifications.base_data import DictField
+
+from edx_notifications.data import (
+    NotificationMessage,
+    NotificationType,
+    NotificationUserMap,
+)
+from edx_notifications import const
+
+
+class SQLNotificationType(models.Model):
+    """
+    Notification Type information
+    """
+
+    # the internal name is the primary key
+    name = models.CharField(primary_key=True, max_length=255)
+
+    class Meta(object):
+        """
+        ORM metadata about this class
+        """
+        app_label = 'edx_notifications'  # since we have this models.py file not in the root app directory
+        db_table = 'edx_notifications_notificationtype'
+
+    def to_data_object(self, options=None):  # pylint: disable=unused-argument
+        """
+        Generate a NotificationType data object
+        """
+
+        return NotificationType(
+            name=self.name,
+        )
+
+    @classmethod
+    def from_data_object(cls, msg_type):
+        """
+        create a ORM model object from a NotificationType
+        """
+
+        obj = SQLNotificationType()
+        obj.load_from_data_object(msg_type)
+        return obj
+
+    def load_from_data_object(self, msg_type):
+        """
+        Hydrate ourselves from a passed in user_map
+        """
+
+        self.name = msg_type.name  # pylint: disable=attribute-defined-outside-init
```

Comment:
Why two methods for this? Why not just instantiate the `SQLNotificationType` with a name directly?